packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTreeComponent.tsx

```diff
@@ -69,106 +67,184 @@ export const ModelsTreeComponent = (props: ModelTreeProps) => {
+    props.viewport?.clearAlwaysDrawn();
+    if (props.iModel) {
       await toggleAllCategories(
         IModelApp.viewManager,
-        iModel,
-        true,
-        viewport,
+        props.iModel,
         false,
-        undefined
+        props.viewport,
+        false
       );
     }
-    viewport?.invalidateScene();
-  }, [viewport, availableModels, iModel]);
+    props.viewport?.invalidateScene();
+  }, [props.viewport, props.availableModels, props.iModel]);
+
+  return (
+    <IconButton
+      className="tree-widget-header-tree-toolbar-icon"
+      icon="icon-visibility-hide-2"
+      title={TreeWidget.translate("hideAll")}
+      onClick={hideAll}
+    />
+  );
+}
+
+export function InvertButtonModelsTree(props: ModelsTreeHeaderButtonProps) {
+  const invert = useCallback(async () => {
+    if (!props.availableModels)
+      return;
+
+    const notViewedModels: string[] = [];
+    const models: string[] = [];
+    props.availableModels.forEach((id: string) => {
+      if (props.viewport?.viewsModel(id)) models.push(id);
+      else notViewedModels.push(id);
+    });
+    await props.viewport?.addViewedModels(notViewedModels);
+    props.viewport?.changeModelDisplay(models, false);
+    props.viewport?.invalidateScene();
+  }, [props.viewport, props.availableModels]);
+
+  return (
+    <IconButton
+      className="tree-widget-header-tree-toolbar-icon"
+      title={TreeWidget.translate("invert")}
+      icon="icon-visibility-invert"
+      onClick={invert}
+    />
+  );
+}
+
+export function View2DButtonModelsTree(props: ModelsTreeHeaderButtonProps) {
+  const [is2dToggleActive, setIs2dToggleActive] = useState<boolean>(false);
+  const [icon2dToggle, setIcon2dToggle] = useState<string>("icon-visibility");
 
   const viewToggle2D = useCallback(async () => {
+    if (!props.availableModels)
+      return;
+
     if (is2dToggleActive) {
-      viewport?.changeModelDisplay(available2dModels, false);
+      props.viewport?.changeModelDisplay(props.availableModels, false);
       setIs2dToggleActive(false);
       setIcon2dToggle("icon-visibility-hide-2");
     } else {
-      await viewport?.addViewedModels(available2dModels);
+      await props.viewport?.addViewedModels(props.availableModels);
       setIs2dToggleActive(true);
       setIcon2dToggle("icon-visibility");
     }
-    viewport?.invalidateScene();
-  }, [is2dToggleActive, viewport, available2dModels]);
+    props.viewport?.invalidateScene();
+  }, [is2dToggleActive, props.viewport, props.availableModels]);
+
+  return (
+    <IconButton
+      className="tree-widget-header-tree-toolbar-icon"
+      icon={icon2dToggle}
+      title={TreeWidget.translate("toggle2DViews")}
+      onClick={viewToggle2D}
+      label={TreeWidget.translate("label2D")}
+    />
+  );
+}
+
+export function View3DButtonModelsTree(props: ModelsTreeHeaderButtonProps) {
+  const [is3dToggleActive, setIs3dToggleActive] = useState<boolean>(false);
```

Comment:
These state variables probably will reset when button is inside dropdown menu (Searbar open) and dropdown menu is reopened

Reply:
fixed